File: server/services/srv-campaign/srv-campaign-client/src/main/java/esthesis/service/campaign/dto/CampaignMemberDTO.java
package esthesis.service.campaign.dto;

import esthesis.common.AppConstants;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
public class CampaignMemberDTO {

  // The identifier of this member, either a device hardware id or a tag name.
  private String identifier;

  // The group this member belongs to.
  private int group;

  // The type of this member, either a device or a tag.
  private AppConstants.Campaign.Member.Type type;
}

File: esthesis-core-ui/src/app/campaigns/dto/campaign-member-dto.ts
import {AppConstants} from "../../app.constants";

export class CampaignMemberDto {
  id?: string;

  // The type of this member, either a device or a tag.
  // See AppConstants.CAMPAIGN.MEMBER_TYPE
  type: string;

  // The identifier for this member, either a device hardware id or a tag name.
  identifier: string;

  // The group in which this member belongs to.
  group: number;

  // @ts-ignore
  constructor(id?: string, type: AppConstants.CAMPAIGN.MEMBER_TYPE, identifier: string,
    groupOrder: number) {
    this.id = id;
    this.type = type;
    this.identifier = identifier;
    this.group = groupOrder;
  }
}
